packages/flutter/test/material/text_field_test.dart

```diff
@@ -711,6 +711,43 @@ void main() {
     expect(controller.selection.extentOffset, eIndex);
   });
 
+  testWidgets('Read only text field', (WidgetTester tester) async {
+    final TextEditingController controller = TextEditingController(text: 'readonly');
+
+    await tester.pumpWidget(
+        overlay(
+          child: TextField(
+            controller: controller,
+            readOnly: true,
+          ),
+        )
+    );
+    // Read only text field cannot open keyboard.
+    await tester.showKeyboard(find.byType(TextField));
+    expect(tester.testTextInput.hasActiveClient, false);
+    await skipPastScrollingAnimation(tester);
+
+    expect(controller.selection.isCollapsed, true);
+
+    // Press the 'd' should not open keyboard.
+    const int dIndex = 3;
+    final Offset dPos = textOffsetToPosition(tester, dIndex);
+    final TestGesture gesture = await tester.startGesture(dPos, pointer: 7);
```

Comment:
Why perform the gesture manually? Would tester.tapAt be sufficient? Or maybe even easier to understand: tester.tap(find.byType(TextField))?

Reply:
I copied from other test, well change

Reply:
fixed